crossfit/backend/torch/hf/memory_curve_utils.py

```diff
@@ -37,7 +41,7 @@ def fit_memory_estimate_curve(
 ) -> LinearRegression:
     print(f"Fitting memory estimate curve for model: {path_or_name}")
 
-    device = next(model.parameters()).device
+    device = "cuda"
```

Comment:
Would it be safer to do
```
try:
    device = next(model.parameters()).device
except AttributeError as e: # unsure if AttributeError is raised
    device = "cuda"
```

Reply:
Actually i want all estimation to happen on `cuda` and if it is not then complain loudly.